### PR TITLE
attach: replace interface with simple type

### PR DESCRIPTION
**Pull request by LK4D4**
ping @cpuguy83 
Also @thaJeztah would you mind to docs review this and #22898 because we decided to merge it and follow-up. Thanks!

### CODE REVIEW

**Comment by cpuguy83** [comment]
LGTM

**Comment by jfrazelle** [comment]
looks good to me

**Comment by thaJeztah** [comment]
For docs, we need the new event added below the diagram here;
https://github.com/docker/docker/blob/master/docs/reference/api/docker_remote_api.md#docker-events

We need to mention it in the API changes;
https://github.com/docker/docker/blob/master/docs/reference/api/docker_remote_api.md#v124-api-changes

In the API docs, add it to the list of events;
https://github.com/docker/docker/blob/master/docs/reference/api/docker_remote_api_v1.24.md#monitor-dockers-events

In the "events" docs, add it to the list of events;
https://github.com/docker/docker/blob/master/docs/reference/commandline/events.md

Same for the man page here;
https://github.com/docker/docker/blob/master/man/docker-events.1.md

Let me know if you need more, or want me to take care of that in (another :smile:) follow up

**Comment by thaJeztah** [comment]
LGTM (thanks @jfrazelle)

**Comment by LK4D4** [comment]
@thaJeztah pls review docs

File: docs/reference/api/docker_remote_api.md [hunk 1]
```diff
@@ -125,6 +127,8 @@ This section lists each version from latest to oldest.  Each listing includes a
 * `POST /containers/(id or name)/start` no longer accepts a `HostConfig`.
 * `POST /images/(name)/tag` no longer has a `force` query parameter.
 * `GET /images/search` now supports maximum returned search results `limit`.
+* Event `detach` is spawned on detaching from container process.
+* Event `exec_detach` is spawned on detaching from exec process.
```
Comment (thaJeztah) [inline docs/reference/api/docker_remote_api.md]: Perhaps keep it similar to the other ones; can you move it up to line 123?

```Markdown
* `GET /events` now supports a `detach` event that is emitted on detaching from container process.
* `GET /events` now supports an `exec_detach ` event that is emitted on detaching from exec process.
```

**Comment by thaJeztah** [comment]
One nit/suggestion, but LGTM otherwise, thanks!

**Comment by LK4D4** [comment]
@thaJeztah pushed fix, thanks!

**Comment by thaJeztah** [comment]
LGTM!

**Comment by WeiZhang555** [comment]
Oh, exec_detach event, this is great! :+1:

**Comment by thaJeztah** [comment]
Thanks for the original PR, @WeiZhang555 apologies for the slightly unconventional way (merge and follow up) to speed up the process :smile:

**Comment by WeiZhang555** [comment]
@thaJeztah No need to apology, I'm quite OK with this way :smile: . 
And I'm glad we can get this feature merged earlier because my another PR depends on it.

I'm in China, so it's true that it could be hard to find a more efficient communication channel.  
I think the `exec_detach` event and docs is great, thanks @LK4D4 and others :+1:

**Comment by thaJeztah** [comment]
Timezones sometimes are a struggle :smile:

**Pull request merged**